### PR TITLE
[data] MapOperator.num_active_tasks should exclude pending actors 

### PR DESCRIPTION
**Pull request by raulchen**
## Why are these changes needed?

`MapOperator.num_active_tasks` should exclude the pending actors. Because
1. `PhysicalOperator.completed` checks `num_active_tasks`. The operator should be considered completed if there are still pending actors.
2. The number of active tasks in the progress bar will be more accurate  to reflect the actual data processing tasks.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/_internal/execution/operators/map_operator.py [hunk 1]
```diff
@@ -414,6 +414,17 @@ def implements_accurate_memory_accounting(self) -> bool:
     def supports_fusion(self) -> bool:
         return self._supports_fusion
 
+    def num_active_tasks(self) -> int:
```
Comment (bveeramani) [inline python/ray/data/_internal/execution/operators/map_operator.py:417]: If I'm understanding correctly, this change makes `len(get_active_tasks()) != num_active_tasks` for map operators. I think that might get confusing.

@raulchen do have any ideas for how we can address the issues while avoiding this discrepancy?

**Review by bveeramani — APPROVED** [review]
LGTM as a short-term fix, but I'm wondering if there are alternative solutions that could avoid the issue I commented about

File: python/ray/data/_internal/execution/operators/map_operator.py (hunk 1 above)
Comment (raulchen) [reply python/ray/data/_internal/execution/operators/map_operator.py:417]: I thought of this as well. It's indeed a bit confusing.
I also thought of separating data and metadata tasks. But that seems overkill.
I think maybe I'll just update the comments to clarify for now.

File: python/ray/data/_internal/execution/interfaces/physical_operator.py [hunk 2]
```diff
@@ -351,12 +351,25 @@ def _get_next_inner(self) -> RefBundle:
...
 
     def num_active_tasks(self) -> int:
         """Return the number of active tasks.
 
+        This method is used for 2 purposes:
+        * Determine if this operator is completed.
+        * Displaying active task info in the progress bar.
+        Thus, the return value can be less than `len(get_active_tasks())`,
+        if some tasks are not needed for the above purposes. E.g., for the
+        actor pool map operator, readiness checking tasks can be excluded
```
Comment (scottjlee) [inline python/ray/data/_internal/execution/interfaces/physical_operator.py:369]: what is "readiness checking task" referring to in this case?

Comment (raulchen) [reply python/ray/data/_internal/execution/interfaces/physical_operator.py:369]: it's `_MapWorker.get_location`.

**Pull request merged**